src/realm/sync/transform.cpp

```diff
@@ -1695,14 +1695,32 @@ DEFINE_MERGE(Instruction::ArrayErase, Instruction::Update)
 DEFINE_MERGE(Instruction::Clear, Instruction::Update)
 {
     using Type = Instruction::Payload::Type;
+    using CollectionType = Instruction::CollectionType;
 
     // The two instructions are at the same level of nesting.
     if (same_path(left, right)) {
-        // TODO: We could make it so a Clear instruction does not win against setting a property or
-        // collection item to a different collection.
-        if (right.value.type != Type::List && right.value.type != Type::Dictionary) {
+        // If both sides are setting/operating on the same type, let them both pass through.
+        // It is important that the instruction application rules reflect this.
+        // If it is not two lists or dictionaries, then the normal last-writer-wins rules will take effect below.
+        if (left.collection_type == CollectionType::List && right.value.type == Type::List) {
+            return;
+        }
+        if (left.collection_type == CollectionType::Dictionary && right.value.type == Type::Dictionary) {
+            return;
+        }
+
```

Comment:
Should this also include `CollectionType::Set`?

Reply:
No. The client and server disallow creating sets in Mixed properties, so there is no sentinel instruction for a Set.

Reply:
Do we have any tests that validate this? I didn't see any in the `test_transform_collections_mixed.cpp` file.

Reply:
I don't think we have a test that explicitly tests that you can't set a mixed to a Set (maybe @jedelbo has one I just can't find), but it is validated in `Obj::set_collection()`